libvast/src/view.cpp

```diff
@@ -388,4 +389,99 @@ data_view to_internal(const type& t, const data_view& x) {
   return caf::visit(v, x, t);
 }
 
+std::shared_ptr<arrow::Scalar>
+to_arrow_scalar(const type& type, const data_view& view) noexcept {
+  VAST_ASSERT(type);
+  VAST_ASSERT(type_check(type, view));
+  auto f = detail::overload{
+    [&]<concrete_type Type>(
+      [[maybe_unused]] const Type& t) -> std::shared_ptr<arrow::Scalar> {
+      if constexpr (detail::is_any_v<Type, map_type, enumeration_type,
+                                     subnet_type>) {
+        // TODO: This is slow because we create a single-element Array and then
+        // use that API to get a scalar for the first element, but I cannot
+        // figure out how to create a scalar for map, enumeraion, or subnet
+        // types directly. -- DL
```

Comment:
since a subnet is mostly just a struct with `address` and `length` fields, we might be able to use the same approach used for `record_type` . For the others, I have no clue :shrug: